libcnb/src/tracing.rs

```diff
@@ -0,0 +1,172 @@
+use libcnb_data::buildpack::Buildpack;
+use opentelemetry::{
+    global,
+    trace::{Span as SpanTrait, Status, Tracer, TracerProvider as TracerProviderTrait},
+    KeyValue,
+};
+use opentelemetry_sdk::{
+    trace::{Config, Span, TracerProvider},
+    Resource,
+};
+use std::{io::BufWriter, path::Path};
+
+// This is the directory in which `BuildpackTrace` stores Open Telemetry File
```

Comment:
I did this purposefully. Clippy will want me to wrap OpenTelemetry in backticks, because it thinks I'm referencing a type. If I fix the casing, I'll also have to add a lint block above it. You're probably right, using the correct casing is more correct.

Reply:
Fixed in [0c18d96](https://github.com/heroku/libcnb.rs/pull/723/commits/0c18d964ad0c9103ee3d5969da998326503ba3d9).

Reply:
Ah I didn't realise that lint would trigger.

Looking now I see that lint has an allowlist - see the "Configuration" section of:
https://rust-lang.github.io/rust-clippy/master/index.html#/doc_markdown

I've opened an upstream PR to add "OpenTelemetry" to the default list:
https://github.com/rust-lang/rust-clippy/pull/11926

In the meantime until that change is on `stable`, we can override the default config using:
https://github.com/heroku/libcnb.rs/blob/main/clippy.toml

Reply:
(Marking as unresolved for now, since CI is failing at the moment due to this lint)